cmd/iam-store.go

```diff
@@ -958,13 +958,12 @@ func (store *IAMStoreSys) PolicyDBUpdate(ctx context.Context, name string, isGro
 	var mp MappedPolicy
 	if !isGroup {
 		if userType == stsUser {
-			var ok bool
-			mp, ok = cache.iamSTSPolicyMap[name]
-			if !ok {
-				// Attempt to load parent user mapping for STS accounts
-				store.loadMappedPolicy(context.TODO(), name, stsUser, false, cache.iamSTSPolicyMap)
-				mp = cache.iamSTSPolicyMap[name]
-			}
+			stsMap := map[string]MappedPolicy{}
+
+			// Attempt to load parent user mapping for STS accounts
+			store.loadMappedPolicy(context.TODO(), name, stsUser, false, stsMap)
```

Comment:
did this on-purpose to avoid cached values, instead rely on on-disk values for latest.